libopenage/curve/tests/curve_types.cpp

```diff
@@ -386,6 +387,86 @@ void curve_types() {
 		TESTEQUALS(c.get(1), 0);
 		TESTEQUALS(c.get(5), 0);
 	}
+
+	{ // array
+		auto f = std::make_shared<event::EventLoop>();
+
+		Array<int, 4> a(f,0);
+		a.set_insert(time::time_t(1), 0, 0);
+		a.set_insert(time::time_t(1), 1, 1);
+		a.set_insert(time::time_t(1), 2, 2);
+		a.set_insert(time::time_t(1), 3, 3);
+		//a = [[0:0, 1:0],[0:0, 1:1],[0:0, 1:2],[0:0, 1:3]]
+
+		auto res = a.get_all(time::time_t(1));
+		TESTEQUALS(res[0], 0);
+		TESTEQUALS(res[1], 1);
+		TESTEQUALS(res[2], 2);
+		TESTEQUALS(res[3], 3);
+
+		Array<int, 4> other(f,0);
+		other[0].last(999);
+		other[1].last(999);
+		other[2].last(999);
+		other[3].last(999);
```

Comment:
1. You are not using `set_last`, so no value is ever inserted.
2. You should not directly access the keyframe container anyway (as mentioned in another comment)